parsers/ocaml.c

```diff
@@ -873,23 +920,33 @@ static void killCurrentState ( void )
 		/* nothing more */
 		break;
 	}
+	return popped;
 }
 
-/* used to prepare tag for OCaml, just in case their is a need to
+/* Keep track of our _true_ line number and file pos,
+ * as the lookahead token gives us false values. */
+static unsigned long ocaLineNumber;
+static fpos_t ocaFilePosition;
+
+/* used to prepare an OCaml tag , just in case their is a need to
  * add additional information to the tag. */
 static void prepareTag (tagEntryInfo * tag, vString const *name, ocamlKind kind)
 {
 	int parentIndex;
 
-	initTagEntry (tag, vStringValue (name));
+	/* Ripped out of read.h initTagEntry, because of line number
+	 * shenanigans.
```

Comment:
I think I tried to do that, but I had some serious compiler errors from what I remember. I think it was specifically related to the linking stage. Don't really recall. But I do know I tried using `initTagEntry()` first. I'll have to take a look at `initTagEntryFull()`.